examples/tv-casting-app/APIs.md

```diff
@@ -810,21 +880,280 @@ let kDesiredEndpointVendorId: UInt16 = 65521;
+                self.connectionStatus = "Connection to Casting Player failed with: \(String(describing: err))"
+            }
         }
-        else
+    }
+
+    let identificationDeclarationOptions: MCIdentificationDeclarationOptions
+    let targetAppInfo: MCTargetAppInfo
+    let connectionCallbacks: MCConnectionCallbacks
+    identificationDeclarationOptions = MCIdentificationDeclarationOptions()
+    targetAppInfo = MCTargetAppInfo(vendorId: kDesiredEndpointVendorId)
+    connectionCallbacks = MCConnectionCallbacks(
+        callbacks: connectionCompleteCallback,
+        commissionerDeclarationCallback: nil
+    )
+    identificationDeclarationOptions.addTargetAppInfo(targetAppInfo)
+
+    let err = selectedCastingPlayer?.verifyOrEstablishConnection(with: connectionCallbacks, identificationDeclarationOptions: identificationDeclarationOptions)
+    if err != nil {
+        self.Log.error("MCConnectionExampleViewModel connect(), MCCastingPlayer.verifyOrEstablishConnection() failed due to: \(err)")
+    }
+}
+```
+
+For a Casting Client to connect to a `CastingPlayer` using the optional
+`CastingPlayer` / Commissioner-Generated Passcode UDC feature, the Casting
+Client may specify optional parameters in the `VerifyOrEstablishConnection`
+function call, handle and then respond to the CastingPlayer's
+CommissionerDeclaration message as follows:
+
+1. In `VerifyOrEstablishConnection` the Casting Client should set the UDC
+   IdentificationDeclaration `CommissionerPasscode` field to true and provide a
+   `ConnectionCallbacks` callback to handle the CastingPlayer's
+   CommissionerDeclaration message.
+1. Upon receiving the CastingPlayer’s CommissionerDeclaration message with
+   PasscodeDialogDisplayed set to true, the Casting Client should prompt the
+   user to input the Passcode displayed on the `CastingPlayer` UX. If the user
+   declines to enter the Passcode, the Casting Client should call
+   `StopConnecting` to alert the `CastingPlayer` that the commissioning attempt
+   was canceled.
+1. The Casting Client should then update the commissioning session's PAKE
+   verifier with the user-entered Passcode using the APIs described in the
+   Initialize the Casting Client section above.
+1. Finally, the Casting Client should call `ContinueConnecting` to send a second
+   IdentificationDeclaration message to the `CastingPlayer` with
+   `CommissionerPasscodeReady` set to true.
+
+On Linux, the Casting Client can connect to a `CastingPlayer`, using the
+optional `CastingPlayer` / Commissioner-Generated Passcode UDC feature, by
+successfully calling `VerifyOrEstablishConnection`, updating the PAKE verifier
+and then calling `ContinueConnecting` on the `CastingPlayer`.
+
+```c
+// VendorId of the Endpoint on the CastingPlayer that the CastingApp desires to interact with after connection
+const uint16_t kDesiredEndpointVendorId = 65521;
+
+void ConnectionHandler(CHIP_ERROR err, matter::casting::core::CastingPlayer * castingPlayer)
+{
```

Comment:
See related comments. We should coalesce these code examples with the pre-existing ones to match it with the actual example code. We should just add inline comments to say what bits are specific to the new flow / old flow. That'll keep the .md file shorter and will make it easier for the reader to see the common parts.

Reply:
merging the examples.